backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -827,54 +826,24 @@ export class ExperimentAssignmentService {
     experimentUser: RequestedExperimentUser,
     context?: string
   ): Promise<[boolean, boolean]> {
-    let userGroup = [];
-    userGroup = Object.keys(experimentUser.workingGroup || {}).map((type: string) => {
+    const userGroup = Object.keys(experimentUser.workingGroup || {}).map((type: string) => {
       return `${type}_${experimentUser.workingGroup[type]}`;
     });
 
-    const globalExcludeSegmentObj = {};
-    const segmentDetails: Segment[] = [],
-      excludedUsers: string[] = [],
-      excludedGroups = [];
-
     if (!context) {
       return [false, false];
     }
 
     // Fetch the global exclude segment for the context
     const globalExcludeSegment = await this.segmentService.getGlobalExcludeSegmentByContext(context);
 
-    if (!globalExcludeSegment) {
-      const error = new Error(`Invalid app context: ${context}`);
-      (error as any).type = SERVER_ERROR.INVALID_APP_CONTEXT;
-      (error as any).httpCode = 400;
-      throw error;
-    }
```

Comment:
Did you intentionally remove this error handling? Without this check, the code will crash when trying to access `globalExcludeSegment.id` if the context is invalid or if `getGlobalExcludeSegmentByContext()` returns `null`.

Reply:
I'm not seeing that change in this pr. Which commit is it from?